source/solvers/navier_stokes_assemblers.cc

```diff
@@ -384,7 +384,8 @@ GLSNavierStokesAssemblerNonNewtonianCore<dim>::assemble_matrix(
           velocity_gradient_x_phi_u_j[j] = velocity_gradient * phi_u_j;
         }
 
-
+      shear_rate_magnitude =
+        shear_rate_magnitude > 1e-3 ? shear_rate_magnitude : 1e-3;
```

Comment:
I am not sure that hard coding this 1e-3 is the best way. Maybe define a parameter or use an already existing parameter.

Reply:
That's a good point, I don't know what would be the best solution for this though. If it's another parameter in the physical properties it means we will have to percolate it everywhere and it will be a bit of a pain. The other solution could be that all rheological model could provide a "minimum shear" parameter and every rheological model could provide it. This should work well.

Reply:
What do you think @CaroleAnneDaunais ?

Reply:
The more I think about it, the more it is best we leave this automatic. Maybe we can merge it as is for now and look into this later.